File: #0628.maximum-product-of-three-numbers.ts
function maximumProduct(nums: number[]): number {

    const length: number = nums.length;
    if (length < 3) {
        return 0;
    }
    nums.sort((n1, n2) => (n1 < n2 ? -1 : 1));
    return Math.max(nums[0] * nums[1] * nums[length - 1], nums[length -3] * nums[length -2] * nums[length -1]);
};

File: #0628.maximum-product-of-three-numbers.cs
public class Solution {
    public int MaximumProduct(int[] nums) {
        
        int length = nums.Length;
        Array.Sort(nums);
        return Math.Max(nums[0] * nums[1] * nums[length - 1], nums[length - 1] * nums[length - 2] * nums[length - 3]);
    }
}
